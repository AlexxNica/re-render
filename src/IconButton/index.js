import React from 'react';
import { Icon } from 'react-native-elements';

export default class IconButton extends React.Component {
    render() {
        let touchableProps = {};

        if (this.props.onPress) {
            touchableProps.onPress = this.props.onPress;
            touchableProps.raised = true;
        }

        return (
            <Icon
                name={this.props.name}
                type='font-awesome'
                color={this.props.color}
                containerStyle={this.props.containerStyle}
                {...touchableProps}/>
        )
    }
}

IconButton.propTypes = {
    name: React.PropTypes.string.isRequired,
    color: React.PropTypes.string,
    onPress: React.PropTypes.func,
    containerStyle: React.PropTypes.object,
};
