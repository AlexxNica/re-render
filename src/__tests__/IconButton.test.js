import React from 'react';
import renderer from 'react-test-renderer';
import IconButton from '../IconButton/index';

it('renders correctly without onPress', () => {
    const tree = renderer.create(
        <IconButton name={'bath'}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with onPress', () => {
    const tree = renderer.create(
        <IconButton name={'bath'} onPress={() => {
        }}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
