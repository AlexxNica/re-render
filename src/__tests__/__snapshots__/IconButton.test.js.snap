// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with onPress 1`] = `
<div
  className="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-transitionDuration-eafdt9 rn-transitionProperty-1i6wzkk rn-userSelect-lrvibr"
  onKeyDown={[Function]}
  onKeyPress={[Function]}
  onKeyUp={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  tabIndex="0"
>
  <div
    className="rn-alignItems-1awozwy rn-backgroundColor-1ikg2hk rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-102gzdx rn-justifyContent-1777fci rn-marginTop-100vyta rn-marginRight-12zb1j4 rn-marginBottom-11mo1y0 rn-marginLeft-162eet2 rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-width-1kz6sp"
    onPress={undefined}
    style={
      Object {
        "borderBottomLeftRadius": "28px",
        "borderBottomRightRadius": "28px",
        "borderTopLeftRadius": "28px",
        "borderTopRightRadius": "28px",
      }
    }
    underlayColor="white"
  >
    <div
      className="rn-backgroundColor-wib322 rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-display-1471scf rn-font-1lw9tu2 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style={
        Object {
          "color": "black",
          "fontFamily": "FontAwesome",
          "fontSize": "24px",
          "fontStyle": "normal",
          "fontWeight": "normal",
        }
      }
    >
      
    </div>
  </div>
</div>
`;

exports[`renders correctly without onPress 1`] = `
<div
  className="rn-alignItems-1awozwy rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim"
  onPress={undefined}
  underlayColor="white"
>
  <div
    className="rn-backgroundColor-wib322 rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-display-1471scf rn-font-1lw9tu2 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style={
      Object {
        "color": "black",
        "fontFamily": "FontAwesome",
        "fontSize": "24px",
        "fontStyle": "normal",
        "fontWeight": "normal",
      }
    }
  >
    
  </div>
</div>
`;
